File: football-game-app/src/services/achievements.js
export const ACHIEVEMENTS = {
  first_win: { id: 'first_win', name: 'First Blood', icon: '🏆', desc: 'Win your first game', coins: 10 },
  win_3: { id: 'win_3', name: 'Hat Trick', icon: '⚽', desc: 'Win 3 games', coins: 25 },
  win_10: { id: 'win_10', name: 'Legend', icon: '👑', desc: 'Win 10 games', coins: 50 },
  streak_3: { id: 'streak_3', name: 'On Fire', icon: '🔥', desc: '3 win streak', coins: 30 },
  streak_5: { id: 'streak_5', name: 'Unstoppable', icon: '💪', desc: '5 win streak', coins: 75 },
  guess_in_3: { id: 'guess_in_3', name: 'Mind Reader', icon: '🧠', desc: 'Guess in 3 questions', coins: 40 },
  guess_in_5: { id: 'guess_in_5', name: 'Quick Thinker', icon: '⚡', desc: 'Guess in 5 questions', coins: 20 },
  speed_demon: { id: 'speed_demon', name: 'Speed Demon', icon: '🏃', desc: 'Win Time Attack with 30s left', coins: 50 },
  perfect_week: { id: 'perfect_week', name: 'Perfect Week', icon: '🌟', desc: 'Complete 7 daily challenges', coins: 100 },
};

export const checkAchievements = (stats) => {
  const unlocked = [];
  
  if (stats.wins === 1 && !stats.achievements?.includes('first_win')) {
    unlocked.push(ACHIEVEMENTS.first_win);
  }
  if (stats.wins === 3 && !stats.achievements?.includes('win_3')) {
    unlocked.push(ACHIEVEMENTS.win_3);
  }
  if (stats.wins === 10 && !stats.achievements?.includes('win_10')) {
    unlocked.push(ACHIEVEMENTS.win_10);
  }
  if (stats.currentStreak === 3 && !stats.achievements?.includes('streak_3')) {
    unlocked.push(ACHIEVEMENTS.streak_3);
  }
  if (stats.currentStreak === 5 && !stats.achievements?.includes('streak_5')) {
    unlocked.push(ACHIEVEMENTS.streak_5);
  }
  
  return unlocked;
};

File: dream/achievements.py
import json
from pathlib import Path
BASE_DIR = Path(__file__).parent
ACHIEVEMENTS_FILE = BASE_DIR / "achievements.json"
ACHIEVEMENTS = {
    "first_win": {"name": "First Victory", "description": "Win your first game", "icon": "🏆"},
    "streak_5": {"name": "Streak Master", "description": "Win 5 games in a row", "icon": "🔥"},
    "streak_10": {"name": "Unstoppable", "description": "Win 10 games in a row", "icon": "⚡"},
    "quick_win": {"name": "Mind Reader", "description": "Win in under 10 questions", "icon": "🧠"},
    "veteran": {"name": "Veteran Player", "description": "Play 50 games", "icon": "🎖️"},
    "perfect_week": {"name": "Perfect Week", "description": "Win 7 games this week", "icon": "💎"}
}
class AchievementTracker:
    def __init__(self, achievements_file=None):
        self.achievements_file = achievements_file or ACHIEVEMENTS_FILE
        self.all_achievements = ACHIEVEMENTS
    def get_unlocked_achievements(self):
        if not self.achievements_file.exists():
            return []
        try:
            with open(self.achievements_file, "r") as f:
                return json.load(f)
        except:
            return []
    def unlock_achievement(self, achievement_id):
        unlocked = self.get_unlocked_achievements()
        if achievement_id not in unlocked:
            unlocked.append(achievement_id)
            try:
                with open(self.achievements_file, "w") as f:
                    json.dump(unlocked, f, indent=2)
            except:
                pass
            return self.all_achievements.get(achievement_id)
        return None
    def check_achievements(self, stats):
        new_achievements = []
        if stats.get("games_won", 0) >= 1:
            ach = self.unlock_achievement("first_win")
            if ach: new_achievements.append(ach)
        current_streak = stats.get("current_streak", 0)
        if current_streak >= 5:
            ach = self.unlock_achievement("streak_5")
            if ach: new_achievements.append(ach)
        if current_streak >= 10:
            ach = self.unlock_achievement("streak_10")
            if ach: new_achievements.append(ach)
        if stats.get("games_played", 0) >= 50:
            ach = self.unlock_achievement("veteran")
            if ach: new_achievements.append(ach)
        return new_achievements
